flatgfa/src/flatbed.rs

```diff
@@ -0,0 +1,191 @@
+use zerocopy::{AsBytes, FromBytes, FromZeroes};
+use crate::memfile::MemchrSplit;
+use std::io::BufRead;
+use atoi::FromRadix10;
+use bstr::BStr;
+use crate::pool::{Pool, Span, Store, HeapStore, FixedStore, Id};
+
+#[derive(Debug, FromZeroes, FromBytes, AsBytes, Clone, Copy)]
+#[repr(packed)]
+pub struct BEDEntry {
+    pub name: Span<u8>,
+    pub start: u64,
+    pub end: u64,
+}
+
+pub struct FlatBED<'a> {
+    pub name_data: Pool<'a, u8>,
+    pub entries: Pool<'a, BEDEntry>,
+}
+
+impl<'a> FlatBED<'a> {
+    /// Get the base-pair sequence for a segment.
+    pub fn get_num_entries(&self) -> usize {
+        self.entries.len()
+    }
+
+    pub fn get_name_of_entry(&self, entry: &BEDEntry) -> &BStr {
+        self.name_data[entry.name].as_ref()
+    }
+
+    pub fn get_intersects(&self, bed: &FlatBED, entry: &BEDEntry) -> Vec<BEDEntry> {
+        self.entries.items()
+            .map(|x| BEDEntry {
+                name: x.1.name,
+                start: if x.1.start < entry.start { entry.start } else { x.1.start },
+                end: if entry.end < x.1.end { entry.end } else { x.1.end },
+            })
+            .filter(|x| bed.get_name_of_entry(entry).eq(self.get_name_of_entry(x)) && x.end > x.start)
```

Comment:
Interesting that there is a string comparison required here. That seems like it is likely expensive. It's not something to do in this PR, but we could consider preprocessing the names so that the comparison here can just use fast integer equality…